### PR TITLE
Speedy: Use fullurl when linking to file logs on CSD log

### PR DESCRIPTION
**Pull request by DannyS712**
No need to write out the full url and place multiple calls to get the server and log url when a parser function does the trick.

### CODE REVIEW

**Comment by Amorymeltzer** [comment]
I'll agree that `fullurl:` is probably more clear to someone editing the wikitext, but this isn't really a speed gain.  The `mw.config` and `mw.util` objects/functions are exported on every page, and while it may technically be slower than a simple string assignment, the difference would be dwarfed by comparison to even how long it would take you to click submit or for Twinkle to display even a single message.  Indeed, once the log is added, it's probably more expensive for the software to compute `fullurl` every time the csd log is loaded compared to using the direct link.  Neither should be a major factor here.

**Comment by DannyS712** [comment]
@Amorymeltzer the primary motivation wasn't the speed, but rather the wikitext - its much harder to read through the source when there are long urls hard-coded in.

**Comment by Amorymeltzer** [comment]
Yes, that's what I meant by
>I'll agree that `fullurl:` is probably more clear to someone editing the wikitext

File: modules/twinklespeedy.js [hunk 1]
```diff
@@ -1634,7 +1634,7 @@ Twinkle.speedy.callbacks = {
 			var extraInfo = '';
 
 			// If a logged file is deleted but exists on commons, the wikilink will be blue, so provide a link to the log
-			var fileLogLink = mw.config.get('wgNamespaceNumber') === 6 ? ' ([' + mw.config.get('wgServer') + mw.util.getUrl('Special:Log', {action: 'view', page: mw.config.get('wgPageName')}) + ' log])' : '';
+			var fileLogLink = mw.config.get('wgNamespaceNumber') === 6 ? ' ([{{fullurl:Special:Log|page=' + mw.config.get('wgPageName') + '}} log])' : '';
```
Comment (Amorymeltzer) [inline modules/twinklespeedy.js]: Needs to properly encode the page title for the URI.

```suggestion
			var fileLogLink = mw.config.get('wgNamespaceNumber') === 6 ? ' ([{{fullurl:Special:Log|page=' + mw.util.wikiUrlencode (mw.config.get('wgPageName')) + '}} log])' : '';
```

`encodeURIComponent` would work too, but using `mw.util` keeps the colon in the namespace intact.

**Pull request merged**